build.gradle

```diff
@@ -49,36 +55,84 @@ dependencies {
 
     // for the RestApiScriptRunner and XmlOutputRunner
     compile('ant:ant:1.7.0')
+
+    compile "org.jenkins-ci.plugins:job-dsl-core:${jobDslVersion}"
+
+    testCompile 'cglib:cglib:3.1' 
+
+    // Jenkins test harness dependencies
+    testCompile 'org.jenkins-ci.main:jenkins-test-harness:2.8'
+    testCompile "org.jenkins-ci.main:jenkins-war:${jenkinsVersion}"
+    testCompile "org.jenkins-ci.main:jenkins-war:${jenkinsVersion}:war-for-test@jar"
+
+    debugXml "ant:ant:1.7.0"
+    debugXml 'org.jenkins-ci.main:jenkins-test-harness:2.8'
+    debugXml "org.jenkins-ci.main:jenkins-war:${jenkinsVersion}"
+    debugXml "org.jenkins-ci.main:jenkins-war:${jenkinsVersion}:war-for-test@jar"
+
+    // Job DSL plugin including plugin dependencies
+    testCompile "org.jenkins-ci.plugins:job-dsl:${jobDslVersion}"
+    testCompile "org.jenkins-ci.plugins:job-dsl:${jobDslVersion}@jar"
+    testCompile 'org.jenkins-ci.plugins:structs:1.1@jar'
+    testCompile 'org.jenkins-ci.plugins:cloudbees-folder:4.4@jar'
+
+    // plugins to install in test instance
+    testPlugins 'org.jenkins-ci.plugins:ghprb:1.31.4'
+    testPlugins 'com.coravy.hudson.plugins.github:github:1.19.0'
+
 }
 
-task cleanLibs(type: Delete) {
-    delete 'lib'
+
+codenarc {
+    toolVersion = '0.25.2'
+    configFile = file('config/codenarc/codenarcRules.groovy')
+    maxPriority2Violations = 49
+    maxPriority3Violations = 99
 }
 
+
+
 task libs(type: Copy) {
     into 'lib'
     from configurations.libs
 }
 
+task cleanLibs(type: Delete) {
```

Comment:
I remember gradle having a default `clean` task. Is this also a default task? Perhaps it's a child, or it's the new `clean`? Basically I want to use conventional targets... I realize you didn't write this originally but it's sticking out to me at the moment.